api/tx_args.go

```diff
@@ -138,6 +138,9 @@ type SendTxArgs struct {
 
 // setDefaults is a helper function that fills in default values for unspecified common tx fields.
 func (args *SendTxArgs) setDefaults(ctx context.Context, b Backend) error {
+	head := b.CurrentBlock().Header()
+	isKIP71 := head.BaseFee != nil
```

Comment:
This function is for setting default values when send tx (or sign), so how about determine whether KIP-71 hard fork or not with next block number?
Maybe similar with `b.ChainConfig().IsKIP71ForkEnabled(new(big.Int).Add(b.CurrentBlock().Number(), big.NewInt(1)))`?